index.js

```diff
@@ -0,0 +1,187 @@
+let getTripBuilder = function(id) {
+  let getTrip = function getTrip() {
+    const successCallback = function(response) {
+      console.log('success!');
+
+      // let listItem = '';
+      // for (let attr in response) {
+      //   listItem += '<li>' + attr + ': ' + response[attr] + '</li>';
+      // }
+      let listItem = '';
+      listItem += '<li><h3>' + response.name + '</h3></li>';
+      listItem += '<li><strong>Continent: </strong>' + response.continent + '</li>';
+      listItem += '<li><strong>About: </strong>' + response.about + '</li>';
+      listItem += '<li><strong>Category: </strong>' + response.category + '</li>';
+      listItem += '<li><strong>Duration in weeks: </strong>' + response.weeks + '</li>';
+      listItem += '<li><strong>Cost: </strong>$' + response.cost + '</li>';
+
+
+      // make reservation
+      let reservationForm =
+      `<form data-id=${id}>
+
+        <div class="input-form">
+          <label for="name"><strong>Name: </strong></label>
+          <input type="text" name="name"></input>
+        </div>
+
+        <div class="input-form">
+          <label for="age"><strong>Age: </strong></label>
+          <input type="number" name="age"></input>
+        </div>
+
+        <div class="input-form">
+          <label for="email"><strong>Email: </strong></label>
+          <input type="text" name="email"></input>
+        </div>
+
+        <input id="submit" class="button blue white-text" type="submit" value="Reserve a trip"></input>
+
+      </form>`
+
+      // what would be displayed
+      $('.wrapper').toggle();
+      $('#trip').html(listItem);
+      $('#reservation').html(reservationForm);
+
+
+    } //end of successCallback
+
+    const failureCallback = function(response) {
+      let generatedHTML = '<p><strong>Something went wrong! Please try again!</strong></p>';
+
+      $('.trip').html(generatedHTML);
+    }; // end of failureCallback
+
+    $.get(tripsURL + `/${id}`, successCallback).fail(failureCallback);
+
+  } // end of getTrip
+  getTrip();
+} // end of getTripBuilder
+
+let makeReservation = function makeReservation(id) {
+
+  const successCallback = function(response) {
+    console.log('POST -> success!');
+    console.log(response);
+
+    let generatedHTML = '<p><strong>Everything went well! Thank you for your reservation!</strong></p>';
+
+    $('#reservation').html(generatedHTML);
+
+  } // end of successCallback
+
+  const failureCallback = function(response) {
+    let generatedHTML = '<p>Something went wrong! Please try again!</p>';
+
+    $('#reservation').html(generatedHTML);
+  }; // end of failureCallback
+
+  let reservationData = $('#reservation form').serialize();
+  let reservationURL = tripsURL + `/${id}` + '/reservations';
+  $.post(reservationURL, reservationData, successCallback).fail(failureCallback);
+
+} // end of makeReservation
+
+
+$(document).ready( function() {
+
+  //get all trips on click of a button
+  $('.trips').on('click', '#trips', function() {
```

Comment:
I very much like the organization of defining all your helper functions and callbacks up above, then binding all the click handlers here in `$(document).ready`. Good work.